pkg/applicationmanager/controllers/applicationbackupschedule.go

```diff
@@ -284,6 +285,28 @@ func getBackupLocationCR(backupSchedule *stork_api.ApplicationBackupSchedule) (*
 	return backuplocationCR, nil
 }
 
+// isPeriodicSchedule - isPeriodicSchedule will check whether the sp has interval type is set.
+func isPeriodicSchedule(schedulePolicyCR *stork_api.SchedulePolicy) bool {
+	return schedulePolicyCR.Policy.Interval != nil
+}
+
+// getLatestSuccessfullbackupTime - will check return the last successful backup for backupschedule.
+// It will check the backup list present in the applicationbackupschedule CR.
+func getLatestSuccessfullbackupTime(backupSchedule *stork_api.ApplicationBackupSchedule) int64 {
+	var lastSuccessfulBackupCreateTime int64
+	for _, statusItems := range backupSchedule.Status.Items {
```

Comment:
If there are all bacups succesful, then how do we guarnatee that this gives us the last one as the list can be in any order? For all successful case, it can bail out in the first success itself and not go till the last one. My thinking was we sort the list based on Time and get the last or the first one (depending on sorting)

Reply:
As discussed the list of backup in the applicationbackupschedule CR is order list and start search from the end of the list for a successful backup.